### PR TITLE
Fixes a few "used but undeclared" dependency problems in starter archetype

### PR DESCRIPTION
**Pull request by jkff**
Example failed log before this PR: https://api.travis-ci.org/jobs/176202650/log.txt?deansi=true

R: @lukecwik

### CODE REVIEW

**Comment by jbonofre** [comment]
CC: @jbonofre

**Comment by jbonofre** [comment]
LGTM

For the record, maybe a Jira would be great.

File: sdks/java/maven-archetypes/starter/pom.xml [hunk 1]
```diff
@@ -79,5 +79,10 @@
       <artifactId>beam-sdks-java-core</artifactId>
       <scope>runtime</scope>
     </dependency>
+
+    <dependency>
+      <groupId>org.slf4j</groupId>
+      <artifactId>slf4j-api</artifactId>
+    </dependency>
```
Comment (lukecwik) [inline sdks/java/maven-archetypes/starter/pom.xml]: The used but undeclared dependency is scoped to runtime.

2016-11-15T23:22:21.460 [WARNING] Used undeclared dependencies found:
2016-11-15T23:22:21.461 [WARNING]    org.slf4j:slf4j-api:jar:1.7.14:runtime

**Comment by jkff** [comment]
Hold on, there's a bunch more dependency problems.

**Comment by jkff** [comment]
Green, PTAL.

**Comment by lukecwik** [comment]
LGTM, merging

**Pull request merged**